### ISSUE TITLE
More tests for invalid cases

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/py-ssz#61]
## What was wrong?

I believe this library could benefit from more test cases that verify decoding detects invalid serializations.

Here are some cases that come to mind

- [ ] offsets that point beyond the end of the data
- [ ] offsets that are less than a previous offset
- [ ] extra bytes on the end of a list of fixed-sized elements
- [ ] extra bytes between the fixed size elements and the variable sized elements in a vector or container
- [ ] not enough bytes or too many bytes in list/vector/container of fixed size elements

## How can it be fixed?

Write these test cases.

**Comment by pipermerriam** [comment]
Running coverage would also be a good way to see which exception cases are not being excercised.

**Comment by jannikluhn** [comment]
Yep, I agree that we should have these kinds of tests. But I think they should be part of the YAML test suite so that we don't multiply the effort of writing these tests across all implementations.
